catroid/src/main/java/org/catrobat/catroid/content/bricks/WriteVariableOnDeviceBrick.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.bricks;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.Constants;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.actions.ScriptSequenceAction;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+
+import java.util.List;
+
+public class WriteVariableOnDeviceBrick extends UserVariableBrick {
+
+	private static final long serialVersionUID = 1L;
+
+	public WriteVariableOnDeviceBrick() {
+	}
+
+	@Override
+	public int getViewResource() {
+		return R.layout.brick_write_variable_on_device;
+	}
+
+	@Override
+	protected int getSpinnerId() {
+		return R.id.write_variable_spinner;
+	}
+
+	@Override
+	public List<ScriptSequenceAction> addActionToSequence(Sprite sprite, ScriptSequenceAction sequence) {
+		if (userVariable == null || userVariable.getName() == null) {
+			userVariable = new UserVariable("NoVariableSet", Constants.NO_VARIABLE_SELECTED);
+			userVariable.setDummy(true);
```

Comment:
from what I see here I would expect that this is going to write `"NoVariableSet", Constants.NO_VARIABLE_SELECTED` to your file but that is not what it does. It has almost the same check in the action to prevent that 

```
		if (userVariable == null || userVariable.isDummy()) {
			return;
		}
```
why even bother setting `.setDummy()` here? you might as well either don't add the action if `userVariable == null || userVariable.getName() == null` is true or just move the check to the action and don't do it here. 

this way it's just unnecessary code obfuscation.